### ISSUE TITLE
location.hash does not update in about:blank

### ISSUE THREAD

**Issue by just-boris** [tmpvar/jsdom#1289]
Unless I do not specify the url, my location hash cannot be updated:

Given a simple test.

```js
const jsdom = require('jsdom').jsdom;
const window = jsdom('<html><head></head><body></body></html>', {}).defaultView;

location.hash = '#test';
console.log(location.hash); // prints ''
```

If I do set a page url, it works fine

```js
const window = jsdom('<html><head></head><body></body></html>', {
  url: 'http://test.page'
}).defaultView;

location.hash = '#test';
console.log(location.hash); // prints '#test'
```

Why there is nessesary to set url to make hash work?

**Comment by domenic** [comment]
Can you test in a browser console on the URL about:blank? That is the default jsdom URL.

**Comment by Sebmaster** [comment]
Works in Chrome. Not sure if Chrome's not up to spec there, or whatwg-url is missing something.

**Comment by just-boris** [comment]
Checked in all available browsers. Chrome, Opera, Safari, IE11 work fine. Firefox fails with error: `Exception... "Unexpected error"  nsresult: "0x8000ffff (NS_ERROR_UNEXPECTED)"`

Firefox already has issue about it: https://bugzilla.mozilla.org/show_bug.cgi?id=1127895

**Comment by domenic** [comment]
Failing test case:

```js
exports["window.location.hash is manipulable even for about:blank (GH-1289)"] = t => {
  const window = jsdom.jsdom("", {
    url: "about:blank"
  }).defaultView;

  t.equals(window.location.hash, "");

  const url = new window.URL("about:blank");
  url.hash = "#baz";

  window.location.hash = "#baz";
  t.equals(window.location.hash, "#baz");
  t.equals(window.location.href, "about:blank#baz");

  t.done();
};
```

This will need to be fixed after #1265 lands.
